pkg/controller/workload/reconciler.go

```diff
@@ -162,18 +161,19 @@ func (r *Reconciler) getSupplyChainsForWorkload(workload *v1alpha1.Workload) (*v
 	}
 
 	supplyChains, err := r.Repo.GetSupplyChainsForWorkload(workload)
-	if err != nil || len(supplyChains) == 0 {
+	if err != nil {
+		return nil, controller.NewUnhandledError(fmt.Errorf("get supply chain by label: %w", err))
```

Comment:
wdyt of

```suggestion
		return nil, controller.NewUnhandledError(fmt.Errorf("get supplychains for workload: %w", err))
```

?

the fact that it uses labels seems to be a detail that would be surfaced by the error message within the method (rather than at this level)

Reply:
We return an unhandled error in one case

Reply:
sorry, wdym? I'm thinking more about the content of the messages - rather than `get supply chain by label`, being more high level with `get supplychains for workload` so that it's up to the lower level (actual `repository.GetSupplyChainsForWorkload`) to let us know the details (the fact that it's by label, or that ... say .. it didn't find a "best match" in the future).

Reply:
ahhh, I cannot read. I'll make the change. I think the entire API is incorrect, but I'll handle that separately.